watertap/unit_models/electroNP_ZO.py

```diff
@@ -0,0 +1,494 @@
+        self.properties_byproduct = self.config.property_package.build_state_block(
+            self.flowsheet().time,
+            doc="Material properties of byproduct stream",
+            **tmp_dict_2,
+        )
+
+        # Create Ports
+        self.add_port("inlet", self.properties_in, doc="Inlet port")
+        self.add_port(
+            "treated", self.properties_treated, doc="Treated water outlet port"
+        )
+        self.add_port(
+            "byproduct", self.properties_byproduct, doc="Byproduct outlet port"
+        )
+
+        # Add isothermal constraints
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Isothermal assumption for treated flow",
+        )
+        def eq_treated_isothermal(b, t):
+            return b.properties_in[t].temperature == b.properties_treated[t].temperature
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Isothermal assumption for byproduct flow",
+        )
+        def eq_byproduct_isothermal(b, t):
+            return (
+                b.properties_in[t].temperature == b.properties_byproduct[t].temperature
+            )
+
+        # Add performance variables
+        self.recovery_frac_mass_H2O = Var(
+            self.flowsheet().time,
+            initialize=0.8,
+            domain=NonNegativeReals,
+            units=pyunits.dimensionless,
+            bounds=(0.0, 1.0000001),
+            doc="Mass recovery fraction of water in the treated stream",
+        )
+        self.removal_frac_mass_comp = Var(
+            self.flowsheet().time,
+            self.config.property_package.solute_set,
+            domain=NonNegativeReals,
+            initialize=0.01,
+            units=pyunits.dimensionless,
+            doc="Solute removal fraction on a mass basis",
+        )
+
+        # Add performance constraints
+        # Water recovery
+        @self.Constraint(self.flowsheet().time, doc="Water recovery equation")
+        def water_recovery_equation(b, t):
+            return b.recovery_frac_mass_H2O[t] * b.properties_in[
+                t
+            ].get_material_flow_terms("Liq", "H2O") == b.properties_treated[
+                t
+            ].get_material_flow_terms(
+                "Liq", "H2O"
+            )
+
+        # Flow balance
+        @self.Constraint(self.flowsheet().time, doc="Overall flow balance")
+        def water_balance(b, t):
+            return b.properties_in[t].get_material_flow_terms(
+                "Liq", "H2O"
+            ) == b.properties_treated[t].get_material_flow_terms(
+                "Liq", "H2O"
+            ) + b.properties_byproduct[
+                t
+            ].get_material_flow_terms(
+                "Liq", "H2O"
+            )
+
+        # default water recovery
+        @self.Constraint(self.flowsheet().time, doc="Default water recovery equation")
+        def default_water_recovery_equation(b, t):
+            return b.recovery_frac_mass_H2O[t] == 1
```

Comment:
I would just set the recovery_frac_mass_H2O variable to an initial value and then fix the variable within this build instead of having this constraint.

Reply:
Yeah, modified.